### DISCUSSION TITLE
Disconnecting when mouse is idle

### DISCUSSION THREAD

**Discussion by mkague** [dbeaver/dbeaver#22835] [Q&A] [answered]
### Description

Hello there! I hope everything is going well.

Since the last update, I have been experiencing frequent disconnections, and I couldn't figure out why. The connection settings are configured to keep the connection alive every 60 seconds.

However, I later discovered that this disconnection behavior occurs when my mouse becomes idle. Regardless of whether a query is running, if the mouse remains inactive, a counter will appear, indicating that the connection will be terminated.

![dbeaver_bad_behavior](https://github.com/dbeaver/dbeaver/assets/160027827/82530c1e-9468-4c94-b1bb-84f6f97393d0)

### DBeaver Version

Community Edition 23.3.4

### Operating System

Windows 11 Pro

### Database and driver

Database: Snowflake

![image](https://github.com/dbeaver/dbeaver/assets/160027827/589ff0ab-fc1d-4077-8576-e69b106c2a53)

Driver: snowflake-jdbc-3.13.30.jar

![image](https://github.com/dbeaver/dbeaver/assets/160027827/f7100cf9-36e7-4fdb-bce9-12b8560e2dae)

### Steps to reproduce

Just stay with the mouse idle.

**Comment by E1izabeth** [comment, marked as answer]
It's an intended behavior. If you don't like it, you can turn this off here
![image](https://github.com/dbeaver/dbeaver/assets/28875055/da4d69ea-af19-4219-a667-47fd648470be)

**Comment by mkague** [comment]
Thank you for clarifying my question.

**Comment by mkague** [comment]
It solved my issue. You are a lifesaver! Thank you 👍
